### ISSUE TITLE
Assigning a new object to self is valid

### ISSUE THREAD

**Issue by ghost** [godotengine/godot#23220]
3.1 Alpha.

```
class Base extends Node:
    
    func test():
        print('BASE')

func test():
    print('SOURCE')

var example = self

func _ready():
    test()
    example = Base.new()
    self = example
    test()
```

This outputs

SOURCE
BASE

This only works if you assign the new class to a variable first. It presents a type error if you try to do this:

```
self = Base.new()
```

Error: "Assigned value type (Base) doesn't match the variable's type (self)"

While the methods change when you assign to self, the variables do not.

```
class Base extends Node:
	var title = 10

var example = self
var title = 20

func _ready():
	self = example
	print(self.title)
```

This will print 20 (instead of 10) which is not in line with the method swapping from the previous example.

**Comment by ThakeeNathees** [comment]
This issue has been fixed, and should be closed

**Comment by akien-mga** [comment]
Indeed, duplicate of #33604 fixed by #35412.
